libcontainer/factory_linux.go

```diff
@@ -249,28 +251,28 @@ type LinuxFactory struct {
 
 func (l *LinuxFactory) Create(id string, config *configs.Config) (Container, error) {
 	if l.Root == "" {
-		return nil, newGenericError(errors.New("invalid root"), ConfigInvalid)
+		return nil, &ConfigError{"invalid root"}
```

Comment:
Do we need `&ConfigError{}`?

Reply:
It's a replacement for "ConfigInvalid" error code, so a caller can distinguish configuration errors from everything else.

Reply:
You're right @AkihiroSuda, it's better not to have it, but I was not able to come with a good solution before. Now I am; please see https://github.com/opencontainers/runc/pull/3056